android/shopify_buy_plugin/src/main/java/com/shopify/unity/buy/AndroidPayCheckoutSession.java

```diff
@@ -61,12 +79,59 @@ public PayCart cartFromUnity(
                 .build();
     }
 
-    private void addAndroidPayFragment(String unityDelegateObjectName, PayCart cart, String countryCode) {
-        UnityAndroidPayFragment payFragment =
-                UnityAndroidPayFragment.newInstance(unityDelegateObjectName, cart, countryCode);
+    private void addPayFragment(PayCart cart, String countryCode, String publicKey, boolean testing) {
+        if (payFragment != null) {
```

Comment:
would be much safer if you use this:
```
Fragment payFragment = UnityPlayer.currentActivity.getFragmentManager().findFragrmentByTag("payFragment");
if (fragment != null) {
  removePayFragment(payFragment);
}
```

So in this way you don't need to rely on local field `payFragment`. There will be only one source of truth, FragmentManager from UnityPlayer.currentActivity